srtcore/buffer_rcv.cpp

```diff
@@ -591,14 +591,24 @@ int CRcvBuffer::getTimespan_ms() const
     if (m_iMaxPosInc == 0)
         return 0;
 
-    const int lastpos = incPos(m_iStartPos, m_iMaxPosInc - 1);
-    // Should not happen if TSBPD is enabled (reading out of order is not allowed).
-    SRT_ASSERT(m_entries[lastpos].pUnit != NULL);
+    int lastpos = incPos(m_iStartPos, m_iMaxPosInc - 1);
+    // Normally the last position should always be non empty
+    // if TSBPD is enabled (reading out of order is not allowed).
+    // However if decryption of the last packet fails, it may be dropped
+    // from the buffer (AES-GCM), and the position will be empty.
+    SRT_ASSERT(m_entries[lastpos].pUnit != NULL || m_entries[lastpos].status == EntryState_Drop);
+    while (m_entries[lastpos].pUnit == NULL)
+    {
+        if (lastpos == m_iStartPos)
```

Comment:
Not sure if the use of two conditions for continuation of the same loop specified in two different instructions look sensible.

Actually if you want to have a numeric range loop, then the condition in the loop should correspond with the step instruction. This way you simply handle the first-found nonempty packet inside the loop already, and if the loop was never broken, you return the zero value then.

Reply:
Changing to:

```c++
    while (m_entries[lastpos].pUnit == NULL && lastpos != m_iStartPos)
    {
        lastpos = decPos(lastpos);
    }
```